drivers/SmartThings/zigbee-power-meter/src/shinasystems/init.lua

```diff
@@ -0,0 +1,56 @@
+-- Copyright 2022 SmartThings
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+local capabilities = require "st.capabilities"
+local ZigbeeDriver = require "st.zigbee"
+local defaults = require "st.zigbee.defaults"
+local constants = require "st.zigbee.constants"
+local clusters = require "st.zigbee.zcl.clusters"
+local ElectricalMeasurement = clusters.ElectricalMeasurement
+local SimpleMetering = clusters.SimpleMetering
+
+local ZIGBEE_POWER_METER_FINGERPRINTS = {
+  { model = "PMM-300Z1" }
+}
+
+local is_shinasystems_power_meter = function(opts, driver, device)
+  for _, fingerprint in ipairs(ZIGBEE_POWER_METER_FINGERPRINTS) do
+      if device:get_model() == fingerprint.model then
+          return true
+      end
+  end
+
+  return false
+end
+
+local do_configure = function(self, device)
+  device:refresh()
+  device:configure()
+end
+
+local device_init = function(self, device)
+  device:set_field(constants.SIMPLE_METERING_DIVISOR_KEY, 1000, {persist = true})
+  device:set_field(constants.ELECTRICAL_MEASUREMENT_DIVISOR_KEY, 1000, {persist = true})
+end
+
+local shinasystems_power_meter_handler = {
+  NAME = "shinasystems power meter handler",
+  lifecycle_handlers = {
+    init = device_init,
+    doConfigure = do_configure,
+  },
+  can_handle = is_shinasystems_power_meter
+}
+
+return shinasystems_power_meter_handler
```

Comment:
this file is identical to the frient handler above, why wouldn't you just include the fingerprints for all of the devices?

Reply:
`frient` and `shinasystems` have different ELECTRICAL_MEASUREMENT_DIVISOR_KEY value.
I considered to merge each sub drivers into one. but It was hard to naming the sub driver.I n fact, I personally prefer to distinguish between drivers by manufacturer, unless there are many devices in fingerprint and therefore code reuse needs to be considered a lot.

Reply:
I missed that difference.